src/target/riscv/batch.c

```diff
@@ -13,58 +13,62 @@
 #define DMI_SCAN_MAX_BIT_LENGTH (DTM_DMI_MAX_ADDRESS_LENGTH + DTM_DMI_DATA_LENGTH + DTM_DMI_OP_LENGTH)
 #define DMI_SCAN_BUF_SIZE (DIV_ROUND_UP(DMI_SCAN_MAX_BIT_LENGTH, 8))
 
+#define BATCH_RESERVED_SCANS 4
```

Comment:
Wouldn't the value of 1 be sufficient?
As far as I understand this is only needed to always be able to insert a `nop` before running the batch.

Reply:
As far as I understand, we really need `4` reserved scans:

- one for the final NOP
- three for the optional bscan bridge tunneling (see here: https://github.com/riscv/riscv-openocd/blob/riscv/src/target/riscv/batch.c#L97)

I have added a comment to make it clearer to the reader.

Reply:
As far as I understand these three additional scans are needed for each tunneled scans and are allocated in the `bscan_ctxt` array. It seems to me [`riscv_add_bscan_tunneled_scan()`](https://github.com/riscv/riscv-openocd/blob/riscv/src/target/riscv/riscv.c#L6322) uses one element of `fields` array and one element of `bscan_ctxt` array, which itself contains extra fields (https://github.com/riscv/riscv-openocd/blob/riscv/src/target/riscv/riscv.h#L327).
Am I missing something?

Reply:
I could have missed the `bscan_ctxt`. I will double-check it again.

Reply:
Hi @en-sc, you're right, the number of the reserved scans should be `1`, and thanks for checking it after me. I mis-read the code in `riscv_batch_run()`. 

I have updated this PR.